src/npr/rules_dorado/08_pycoqc.smk

```diff
@@ -0,0 +1,34 @@
+'''
+pycoqc
+'''
+
+# define source and target pattern
+source = sample_dat + ".seqsum"
+target_html = sample_qc + "_pycoqc.html"
+target_json = sample_qc + "_pycoqc.json"
+logpat = sample_log + "_pycoqc.log"
+bchpat = sample_bch + "_pycoqc.tsv"
+
+rule pycoqc_final:
+    input: expand_project_path(target_html)
+    output: touch("flags/08_pycoqc.done")
+
+rule pycoqc:
+    input: 
+        seqsum=source
+    output:
+        html=target_html,
+    wildcard_constraints:
+        # exclude all sample files that end on ".align.bam" (already aligned) 
+        sample_name = r'(?!.*\.align_pycoqc\.json$).*',
+        sample_id="[0-9]{2}L[0-9]{6}"
+    log:
+        file=logpat,
+        json=target_json
+    benchmark:
+        bchpat
+    shell:'''
+        touch {output.html}  # since pycoqc may fail
+        pycoQC --summary_file {input.seqsum} \
+        {config[pycoQc][pycoQc_opts]} -o {output.html} -j {log.json} >> {log.file} 2>&1 || true
```

Comment:
I would prefer to keep it in - and ensure that {output.html} is always there - if empty. At lease in my test cases with few reads pycoqc failed repeatedly. In fact, this was more important in rule 06_align.smk (which includes a bam file) - and I just copied it over. 
Perhaps the reporting should be improved, but I don't want snakemake to fail just because of pycoQC.